oracle/cpi.yml

```diff
@@ -0,0 +1,109 @@
+---
+- type: replace
+  path: /releases/-
+  value:
+    name: bosh-oracle-cpi
+    url: file://((local_oracle_cpi_release)) 
+
+- type: replace
+  path: /resource_pools/name=vms/stemcell?
+  value:
+    url: file://((local_oracle_light_stemcell))
+
+# Configure instance shapes 
+- type: replace
+  path: /resource_pools/name=vms/cloud_properties?
+  value:
+    availability_domain: ((ad))
+    instance_shape: VM.Standard1.2
+
+# Configure VCN and subnet
+- type: replace
+  path: /networks/name=default/subnets/0/cloud_properties?
+  value:
+    vcn: ((vcn))
+    subnet_name: ((subnet))
+
+# Enable registry job
+- type: replace
+  path: /instance_groups/name=bosh/jobs/-
+  value:
+    name: registry
+    release: bosh
+
+- type: replace
+  path: /instance_groups/name=bosh/properties/registry?
+  value:
+    address: ((internal_ip))
+    host: ((internal_ip))
+    db: # todo remove
+      host: 127.0.0.1
+      user: postgres
+      password: ((postgres_password))
+      database: bosh
+      adapter: postgres
+    http:
+      user: registry
+      password: ((registry_password))
+      port: 25777
+    username: registry
+    password: ((registry_password))
+    port: 25777
+
+# Oracle Cloud Infrastructure properties for BOSH
+- type: replace
+  path: /instance_groups/name=bosh/properties/oci?
+  value: &cpi_conf
+    user: ((user))
+    fingerprint: ((fingerprint))
+    tenancy: ((tenancy))
+    region: ((region))
+    compartment: ((compartment))
+    apikey: ((apikey))
+    cpikey: ((cpikeys.private_key))
```

Comment:
Interesting. Can you tell me (or link me elsewhere) about why the CPI needs a private key to the VM?

Reply:
It's used by by the disk attachment implementation.  Persistent storage in our cloud is available over iSCSI, and one needs to run additional commands on the target vm to establish a iSCSI session etc.  to make disks available to that node.  CPI uses its own ssh key pair to running those commands remotely over ssh. 

@cppforlife and I have talked about enhancing bosh-agent to support "iscsi" as  first class disk type, but that isn't there yet. I understand there is PR coming from another vendor (Softlayer?) that we can perhaps generalize to satisfy other IaaS providers , but until then we need to